pkg/steps/multi_stage/multi_stage.go

```diff
@@ -149,6 +150,7 @@ func newMultiStageTestStep(
 		pre:              ms.Pre,
 		test:             ms.Test,
 		post:             ms.Post,
+		dnsConfig:        ms.DNSConfig,
```

Comment:
If there is a line in particular I'm not confident in for this PR, it's this one.
Is the MultiStageStepConfigurationLiteral the result of all of the connected workflow, chains, and refs?
If so, this will just pass it along to the step object that will in turn safely overwrite a ref if and only if it's unset.

Reply:
If not, then there is sure to be logic that needs to be added to prevent overwriting a sub-chain with a chain and that chain with the workflow and so on.